kolibri/content/management/commands/importcontent.py

```diff
@@ -150,6 +150,7 @@ def _transfer(self, method, channel_id, path=None, node_ids=None, exclude_node_i
                                 file_dl_progress_update(length)
 
                     file_checksums_to_annotate.append(f.id)
+                    annotation.set_availability(channel_id, file_checksums_to_annotate)
```

Comment:
This is calling annotation every time in the loop, rather than waiting until the end of the loop to annotate.

This is not the right way to fix it, I think the fix should be on the frontend, to have parallel behaviour to an import being properly completed (where you click the 'close' button on the notification which then refreshes the channel information from the frontend).

Reply:
This will require the 'cancelling' state to then move to 'cancelled' when the cancel has completed (including the annotation). @aronasorman can you advise on how to use Iceqube to manage this?